lib/battleship/game.ex

```diff
@@ -32,86 +35,44 @@ defmodule Battleship.Game do
     {:error, "Game is full"}
   end
 
-  def insert_boat(boat, %{player1: %{boats: boats_list}} = state, pid)
-      when state.player1.pid == pid do
-    # if is_position_valid?(boat, boats_list) do
-    #   state =
-    #     state
-    #     |> put_in([:player1, :boats], state.player1.boats ++ boat)
+  def insert_boat({boat, pid}, %{player1: %{pid: pid1}, player2: %{pid: pid2}, mode: :setting} = state) do
+    {atom, other} = player(pid, pid1, pid2)
+    if atom == :error do
+      {atom, other}
+    else
+      if Operationsgame.is_position_valid?(boat, get_in(state, [atom, :boats]), state.available_boats) do
+        state =
+          state
+          |> put_in([atom, :boats], get_in(state, [atom, :boats]) ++ [boat])
 
-    #   if all_boats_in_player?(get_in(state, [:player1, :boats])) do
-    #     if all_boats_in?(get_in(state, [:player1, :boats]), get_in(state, [:player2, :boats])) do
-    #       state =
-    #         state
-    #         |> Map.put(:mode, :p1)
+        if Operationsgame.all_boats_set?(get_in(state, [atom, :boats]), state.available_boats) do
+          if Operationsgame.all_boats_set?(get_in(state, [other, :boats]), state.available_boats) do
+            state =
+              state
+              |> Map.put(:mode, :p1)
 
-    #       {:ready, state}
-    #     else
-    #       {:full, state}
-    #     end
-    #   else
-    #     {:ok, state}
-    #   end
-    # else
-    #   {:error, "Position no valid"}
-    # end
+            {:ready, state}
```

Comment:
Alejandro and I have set that ready is when one player has set all his boats and is waiting for the other player to set his ones. I see that you call that 'full', and 'ready' when both players have set their boats, so we should talk about this to have the code coherent.

Reply:
Please let me know when you resolve this because I didn't know about this concepts.

Reply:
Full and ready are just comments to indicate what you have said ahead Sergio. There is no problem to change them. Obviously due to we have not talked yet about the communication of the processes, it was impossible to know what the other person had written. Tomorrow we will discuss it.